src/coreclr/vm/interpexec.cpp

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "interpexec.h"
+
+#ifdef FEATURE_INTERPRETER
+
+thread_local InterpThreadContext *t_pThreadContext = NULL;
+
+InterpThreadContext* InterpGetThreadContext()
+{
+    InterpThreadContext *threadContext = t_pThreadContext;
+
+    if (!threadContext)
+    {
+        threadContext = new InterpThreadContext;
+        // FIXME valloc with INTERP_STACK_ALIGNMENT alignment
```

Comment:
According to doc, `valloc` is an obsolete function and it aligns on page boundary. We should use `aligned_alloc` instead (once we decide to fix this "FIXME").

Reply:
I wasn't even aware there was a function `valloc` in the C libraries. I meant that instead of using the malloc allocator we should allocate directly from OS with VirtualAlloc/mmap given this is a large chunk of memory. But I'm not yet familiar with the PAL stuff so I just added the comment.

Reply:
Ah, got it.